boards/tenstorrent/tt_blackhole/tt_blackhole_fixed_partitions.dtsi

```diff
@@ -14,101 +20,171 @@ partitions {
 
-	memfw: partition@42000 {
-		label = "memfw";
-		reg = <0x42000 DT_SIZE_K(16)>;
-		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_init.bin";
+	flshinfo@19000 {
+		label = "flshinfo";
+		reg = <0x19000 DT_SIZE_K(4)>;
+		binary-path =
+			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
 		read-only;
 	};
 
-	ethsdreg: partition@46000 {
-		label = "ethsdreg";
-		reg = <0x46000 DT_SIZE_K(4)>;
-		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fwreg.bin";
+	/* Remote chip firmware images */
+	ethfw@1a000 {
+		label = "ethfw";
+		reg = <0x1a000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_erisc.bin";
 		read-only;
 	};
 
-	ethsdfw: partition@47000 {
+	memfw@2a000 {
+		label = "memfw";
+		reg = <0x2a000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_init.bin";
+		read-only;
+	};
+
+	ethsdfw@3a000 {
 		label = "ethsdfw";
-		reg = <0x47000 DT_SIZE_K(20)>;
+		reg = <0x3a000 DT_SIZE_K(64)>;
 		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fw.bin";
 		read-only;
 	};
 
-	bmfw: partition@4c000 {
-		label = "bmfw";
-		reg = <0x4c000 DT_SIZE_K(68)>;
-		binary-path = "$BUILD_DIR/dmc/zephyr/zephyr.signed.bin";
-		read-only;
+	/* Update firmware image for the DMC */
+	dmfw: partition@4a000 {
+		label = "dmfw";
+		reg = <0x4a000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/dmc/zephyr/zephyr.bin";
 	};
 
-	flshinfo: partition@59000 {
-		label = "flshinfo";
-		reg = <0x59000 DT_SIZE_K(4)>;
-		binary-path =
-			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
-		read-only;
+	/*
+	 * Image trailer for the DMC firmware update. Indicates to MCUBoot that
+	 * the image is valid and should be booted in test mode.
+	 */
+	dmfw_trailer: partition@c9000 {
+		label = "dmfw_trailer";
+		reg = <0xc9000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
 	};
 
-	boardcfg: partition@5a000 {
-		label = "boardcfg";
-		reg = <0x5a000 DT_SIZE_K(4)>;
-		binary-path = "$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/read_only.bin";
-		read-only;
+	/*
+	 * This partition is used for the primary firmware image.
+	 * it is loaded by MCUBoot. It is tagged as
+	 * mainimg in tt-boot-fs, but isn't used by the bootrom.
+	 */
+	slot0_partition: partition@ca000 {
+		label = "mainimg";
+		reg = <0xca000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/smc/zephyr/zephyr.signed.bin";
 	};
 
 	/*
-	 * FIXME: The address of this node triggers a warning that we should try to fix:
-	 * Warning (unique_unit_address_if_enabled):
-	 *  /soc/spi@80070000/eeprom@0/partitions/partition@10000000
-	 * duplicate unit-address (also used in node
-	 *   /soc/spi@80070000/eeprom@0/partitions/failover_partition@10000000)
+	 * This partition defines a binary blob that indicates the
+	 * main image should be booted in test mode by mcuboot.
 	 */
-	failover: failover_partition@10000000 {
+	main_test: partition@149000 {
+		label = "maintail";
+		reg = <0x149000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
+	};
+
+	/*
```

Comment:
Probably doesn't need to happen right now, but as discussed, we should try to move these bits closer the start of flash in order to potentially facilitate a write-protected area.

Reply:
Done